### DISCUSSION TITLE
How to fetch supercharging history?

### DISCUSSION THREAD

**Discussion by mrgaolei** [timdorr/tesla-api#618] [Q&A]
I Tried GET `https://owner-api.teslamotors.com/bff/v2/mobile-app/charging/history`

but 404 returned.

**Comment by core-hacked** [comment]
@mrgaolei bff is probably the `ownership.tesla.com` api not owner-api.
Ownership is undocumented :/

**Comment by jankratochvil** [comment]
The URL I had noted is: `https://owner-api.teslamotors.com/api/1/vehicles/1234vehicleid56789/charge_history`
One needs to also provide header: `X-Tesla-User-Agent: TeslaApp/4.10.0`
Note sure if this header is required or not: `Content-Type: application/json; charset=UTF-8`
It works in: `https://github.com/tdorssers/TeslaPy`

**Comment by matfroh** [comment]
I tried doing it here, but cannot format the answer. Maybe that helps someone further: https://github.com/timdorr/tesla-api/issues/445#issuecomment-1121539217

**Reply by jankratochvil** [reply]
In such case your request is wrong - Tesla server then sends an empty reply.
I have implemented the "charge_history" functionality into Perl Tesla client and it works for me there ("history"):
 - https://github.com/stevieb9/tesla-api/pull/12
 - https://github.com/stevieb9/tesla-vehicle/pull/8
